docs/design.md

```diff
@@ -155,88 +155,35 @@ Kata Container support has to be enabled explicitly via a [storage
-interface](https://github.com/container-storage-interface/spec/blob/master/spec.md#identity-service-rpc).
-
-### Node Registry Server
-
-When the PMEM-CSI driver runs in _Controller_ mode, it starts a gRPC
-server on a given endpoint(_-registryEndpoint_) and serves the
-[RegistryServer](/pkg/pmem-registry/pmem-registry.proto) interface. The
-driver(s) running in _Node_ mode can register themselves with node
-specific information such as node id,
-[NodeControllerServer](#node-controller-server) endpoint, and their
-available persistent memory capacity.
-
-### Master Controller Server
-
-This gRPC server is started by the PMEM-CSI driver running in
-_Controller_ mode and serves the
-[Controller](https://github.com/container-storage-interface/spec/blob/master/spec.md#controller-service-rpc)
-interface defined by the CSI specification. The server responds to
-CreateVolume(), DeleteVolume(), ControllerPublishVolume(),
-ControllerUnpublishVolume(), and ListVolumes() calls coming from
-external-provisioner() and external-attacher() sidecars. It
-forwards the publish and unpublish volume requests to the appropriate
-[Node controller server](#node-controller-server) running on a worker
-node that was registered with the driver.
-
-### Node Controller Server
-
-This gRPC server is started by the PMEM-CSI driver running in _Node_
-mode and implements the
-[ControllerPublishVolume](https://github.com/container-storage-interface/spec/blob/master/spec.md#controllerpublishvolume)
-and
-[ControllerUnpublishVolume](https://github.com/container-storage-interface/spec/blob/master/spec.md#controllerunpublishvolume)
-methods of the [Controller
-service](https://github.com/container-storage-interface/spec/blob/master/spec.md#controller-service-rpc)
-interface defined by the CSI specification. It serves the
-ControllerPublishVolume() and ControllerUnpublish() requests coming
-from the [Master controller server](#master-controller-server) and
-creates/deletes persistent memory devices.
-
-### Node Server
-
-This gRPC server is started by the driver running in _Node_ mode and
-implements the [Node
-service](https://github.com/container-storage-interface/spec/blob/master/spec.md#node-service-rpc)
-interface defined in the CSI specification. It serves the
-NodeStageVolume(), NodeUnstageVolume(), NodePublishVolume(), and
-NodeUnpublishVolume() requests coming from the Container Orchestrator
-(CO).
+## Dynamic provisioning of local volumes
+
+Traditionally, Kubernetes expects that a driver deployment has a
+central component, usually implemented with the `external-provisioner`
+and a custom CSI driver component which implements volume creation.
+That central component is hard to implement for a CSI driver that
+creates volumes locally on a node.
+
+PMEM-CSI solves this problem by deploying `external-provisioner`
+alongside each node driver and enabling ["distributed
+provisioning"](https://github.com/kubernetes-csi/external-provisioner/issues/487):
```

Comment:
Wouldn't it be better to link to the documentation if any  instead of the issue.

Reply:
Yes, now that that it is merged. Will change it.

Reply:
Fixed.